test/Tests/src/Data/Vector_Spec.enso

```diff
@@ -752,7 +754,11 @@ spec =
             [True, False, 'a'].pretty . should_equal "[True, False, 'a']"
             [Foo.Value True].pretty . should_equal "[(Foo.Value True)]"
 
-    type_spec "Use Vector as vectors" (x -> x)
-    type_spec "Use Array as vectors" (x -> x.to_array)
+    type_spec "Use Vector as vectors" identity
+    type_spec "Use Array as vectors" (v -> v.to_array)
+    type_spec "Use Java ArrayList as vectors" v->
```

Comment:
Please note that [in the original bug report I had a follow up comment](https://github.com/enso-org/enso/issues/6609#issuecomment-1539075618) which also noted that the same issue happens if we get hold of an instance of `ArraySlice`. Which can be easily done by calling `to_array` on a simple vector that has been `slice`d.

I'd like to know if this fix fixes that case as well - so it would be good to add a test case for this too.

Reply:
And for completeness, we should also have at least a small and simple test case for `ArrayOverBuffer` and `ArrayProxy` as well, which are kind-of siblings to `ArraySlice` (in that they are alternative underlying implementations for an `Array` in Enso.

Reply:
(although I'm not certain if stuff like `Array_Proxy` must have the full `Array` API available - I guess it could be good for it to work, but we can (and should) always wrap it with a `Vector`, so I'd be fine with the API not being present - but it would be good to have a test that demonstrates if the API is available or not and that if it is not available, using it at least does not crash the interpreter as it used to (which I assume your fix ensured, but this test would be helpful to demonstrate that).

Reply:
> And for completeness, we should also have at least a small and simple test case for `ArrayOverBuffer` and `ArrayProxy` as well, which are kind-of siblings to `ArraySlice` (in that they are alternative underlying implementations for an `Array` in Enso.

I don't think those two will work but I also don't think this belongs to this PR (to keep it short and reviewable). I think I raised this in the PR that did the Array/Vector integration as well.

Reply:
I don't think there is any reason why any Truffle "array-like" implementation shall not work: [24749fd](https://github.com/enso-org/enso/pull/6642/commits/24749fdf3c839c0530a355ad5a6d874bc6e19c29)